src/EditorFeatures/Core/Wrapping/AbstractParameterWrappingCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,137 @@
+        {
+            var span = context.Span;
+            if (!span.IsEmpty)
+            {
+                return;
+            }
+
+            var position = span.Start;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var token = root.FindToken(position);
+
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            var generator = document.GetLanguageService<SyntaxGenerator>();
+
+            var declaration = token.Parent.Ancestors().FirstOrDefault(n => GetApplicableList(n) != null);
+            if (declaration == null)
+            {
+                return;
+            }
+
+            var listSyntax = GetApplicableList(declaration);
+            // Make sure we don't have any syntax errors here.  Don't want to format if we don't
+            // really understand what's going on.
+            if (listSyntax.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
+            {
+                return;
+            }
+
+            if (!PositionIsApplicable(root, position, declaration, listSyntax))
+            {
+                return;
+            }
+
+            var listItems = GetListItems(listSyntax);
+            if (listItems.Count <= 1)
+            {
+                // nothing to do with 0-1 items.  Simple enough for users to just edit
+                // themselves, and this prevents constant clutter with formatting that isn't
+                // really that useful.
+                return;
+            }
+
+            // For now, don't offer if any item spans multiple lines.  We'll very likely screw up
+            // formatting badly.  If this is really important to support, we can put in the effort
+            // to properly move multi-line items around (which would involve properly fixing up the
+            // indentation of lines within them.
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var item in listItems)
+            {
+                if (item == null ||
+                    item.Span.IsEmpty ||
+                    !sourceText.AreOnSameLine(item.GetFirstToken(), item.GetLastToken()))
+                {
+                    return;
+                }
+            }
+
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var computer = new CodeActionComputer(this, document, options, listSyntax, listItems);
+            var codeActions = await computer.DoAsync(cancellationToken).ConfigureAwait(false);
+
+            context.RegisterRefactorings(codeActions);
+        }
+
+        private static ImmutableArray<CodeAction> SortActionsByMRU(ImmutableArray<CodeAction> codeActions)
```

Comment:
nit: `MRU` might not be obvious to everybody. Consider at least adding a comment spelling it out if you prefer to keep this name.

Reply:
sure.

Reply:
(will do this).